business/checkers/k8sgateways/multi_match_checker.go

```diff
@@ -0,0 +1,98 @@
+package k8sgateways
+
+import (
+	"fmt"
+
+	k8s_networking_v1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	"github.com/kiali/kiali/models"
+)
+
+type MultiMatchChecker struct {
+	K8sGateways []*k8s_networking_v1alpha2.Gateway
+}
+
+const (
+	K8sGatewayCheckerType = "k8sgateway"
+)
+
+// Check validates that no two gateways share the same host+port combination
+func (m MultiMatchChecker) Check() models.IstioValidations {
+	validations := models.IstioValidations{}
+
+	for _, g := range m.K8sGateways {
+		gatewayRuleName := g.Name
+		gatewayNamespace := g.Namespace
+
+		// With addresses
+		for _, address := range g.Spec.Addresses {
+			duplicate, _ := m.findMatchIP(address, g.Name)
+			if duplicate {
+				// The above is referenced by each one below..
+				currentHostValidation := createError(gatewayRuleName, gatewayNamespace, address.Value, "addresses/value")
```

Comment:
here I would suggest to have different error code and message, something like: "More then one K8s Gateway with same address type value", wdyt?

Reply:
Good idea, I've added a new error code. Thanks!